addon/components/frost-expand.js

```diff
@@ -0,0 +1,139 @@
+/**
+ * Component definition for the frost-expand component
+ */
+import Ember from 'ember'
+const {observer} = Ember
+import layout from '../templates/components/frost-expand'
+import computed, {readOnly} from 'ember-computed-decorators'
+import Component from './frost-component'
+import {PropTypes} from 'ember-prop-types'
+
+export default Component.extend({
+  // == Dependencies ==========================================================
+
+  // == Keyword Properties ====================================================
+
+  layout,
+  tagName: 'div',
+  classNameBindings: ['cssStateClass'],
```

Comment:
There is a neat syntax for classNameBindings that actually lets you specify true/false values 'condition:trueValue:falseValue', so I'd suggest simplifying this to:

`classNameBindings: ['expanded:expanded:collapsed']`